src/main/java/com/dungeoncrawler/Javiarenka/character/Monster.java

```diff
@@ -35,28 +35,29 @@ public void setDamageStrength(int damageStrength) {
     @Override
     public String attack(Character hero) {
         String message;
-        String killMessage = "";
         Hero attackedHero = (Hero) hero;
+        int dealtDamage = (damageStrength - attackedHero.getEquippedArmor().getDamageReduction());
 
         Random random = new Random();
         int hitPossibility = random.nextInt(100);
 
         if (!attackedHero.isAlive()) {
-            message = "Monster " + getName() + " can't attack Hero " + attackedHero.getName() + " because he is dead!";
-        } else if (hitPossibility <= attackedHero.getEquippedArmor().getChanceToHitReduction()) {
-            message = "Monster " + getName() + " missed attack on Hero " + attackedHero.getName() + "!";
-        } else if (attackedHero.getEquippedArmor().getDamageReduction() < damageStrength) {
-            attackedHero.setHp(attackedHero.getHp() + attackedHero.getEquippedArmor().getDamageReduction() - damageStrength);
-            message = "Monster " + getName() + " attacked Hero " + attackedHero.getName() + " and dealt " +
-                    (damageStrength - attackedHero.getEquippedArmor().getDamageReduction()) + " damage. ";
-            if (attackedHero.getHp() < 1) {
-                attackedHero.setAlive(false);
-                attackedHero.setHp(0);
-                killMessage += "Hero " + attackedHero.getName() + " has been killed by Monster " + getName() + " !";
-            }
+            return String.format("Monster %s can't attack Hero %s because he is dead!", getName(), attackedHero.getName());
         } else {
-            message = "Monster " + getName() + " is to weak to attack Hero " + attackedHero.getName() + "!";
+            if (hitPossibility <= attackedHero.getEquippedArmor().getChanceToHitReduction()) {
+                return String.format("Monster %s missed attack on Hero %s!", getName(), attackedHero.getName());
+            } else if (attackedHero.getEquippedArmor().getDamageReduction() < damageStrength) {
+                attackedHero.setHp(attackedHero.getHp() - dealtDamage);
```

Comment:
W sumie to często będziemy zadawać obrażenia różnym kreaturom, to może zamiast za każdym razem pisać 
`attackedHero.setHp(attackedHero.getHp() - dealtDamage)`
zrobimy nową metodę:
```
dealDamage(int damage){
attackedHero.setHp(attackedHero.getHp() - damage);
} 
```
i bedziemy zawsze jej używać. Mniej kodu.

Reply:
👍

Reply:
racja tylko że ta metoda musi przyjąć jeszcze herosa:
    private void dealDamage(Hero hero, int dealtDamage) {
        hero.setHp(attackedHero.getHp() - dealtDamage);
    }

ale zastanawiam się tylko czy nie lepiej tę metodę zrobić w klasie Character bo moglibyśmy też ją użyć w klasie Hero bez kolejnego powielenia kodu?

Reply:
Póki jesteśmy w klasie dziedziczącej po Character, to nie musimy podawać herosa. Wystarczy nam:
```private void dealDamage(int dealtDamage) {
this.setHp(this.getHp() - dealtDamage);
}```